core/contract/src/main/java/org/eclipse/dataspaceconnector/contract/negotiation/ProviderContractNegotiationManagerImpl.java

```diff
@@ -333,21 +331,25 @@ private boolean processProviderOffering(ContractNegotiation negotiation) {
                 .correlationId(negotiation.getCorrelationId())
                 .build();
 
+        negotiation.lock();
+        negotiationStore.save(negotiation);
+
         //TODO protocol-independent response type?
         dispatcherRegistry.send(Object.class, contractOfferRequest, () -> null)
                 .whenComplete((response, throwable) -> {
+                    var contractNegotiation = negotiationStore.find(negotiation.getId());
                     if (throwable == null) {
-                        negotiation.transitionOffered();
-                        negotiationStore.save(negotiation);
-                        observable.invokeForEach(l -> l.providerOffered(negotiation));
+                        contractNegotiation.transitionOffered();
```

Comment:
I think `contractNegotiation` should be checked for null here (like in the `TransferProcessManagerImpl`), otherwise might lead to a NullPointerException. Same for other methods in both *NegotiationManagers.

Reply:
Yes, you're right, I didn't do that because it was already done in #642, I didn't want to replicate too much logic here since I'll have enough conflicts to fix when it will be the time for rebase

Reply:
Makes sense, I missed that it was done on the other branch.